apps/src/code-studio/components/progress/ViewAsToggle.jsx

```diff
@@ -19,31 +19,30 @@ class ViewAsToggle extends React.Component {
   };
 
   componentDidMount() {
-    // Upon loading, toggle hide-as-student appropriately (this is so that if we
-    // load a page with ?viewAs=Participant we still hide stuff)
-    const {viewAs} = this.props;
-    $('.hide-as-student').toggle(viewAs === ViewType.Instructor);
+    this.toggleHideAsStudent(this.props.viewAs);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.viewAs !== this.props.viewAs) {
+      this.toggleHideAsStudent(nextProps.viewAs);
+    }
   }
 
+  toggleHideAsStudent = viewAs => {
+    // Toggle hide-as-student appropriately (this is so that if we
+    // load a page with ?viewAs=Participant we still hide teacher-only content)
+    $('.hide-as-student').toggle(viewAs === ViewType.Instructor);
+  };
+
   onChange = viewType => {
-    const {changeViewType} = this.props;
+    const {changeViewType, logToFirehose} = this.props;
 
     updateQueryParam('viewAs', viewType);
 
-    if (viewType === ViewType.Participant && queryParams('user_id')) {
```

Comment:
Just checking that it's ok that we're no longer checking for 'user_id'?

Reply:
I tested this case manually and the page seemed to behave as expected, I'm not sure what the intended purpose of this is statement was. From the comment it looks like the intent was not to change the UI because the page was going to reload anyway, if that was the purpose, removing this should be fine.